protocol/communities_messenger_token_permissions_test.go

```diff
@@ -529,8 +529,30 @@ func (s *MessengerCommunitiesTokenPermissionsSuite) TestEditSharedAddresses() {
 	s.Require().Equal(alicesRevealedAccounts[0].Address, aliceAddress2)
 	s.Require().Equal(true, alicesRevealedAccounts[0].IsAirdropAddress)
 
+	request := &requests.EditSharedAddresses{CommunityID: community.ID(), AddressesToReveal: []string{aliceAddress1}, AirdropAddress: aliceAddress1}
+
+	signingParams, err := s.alice.GenerateJoiningCommunityRequestsForSigning(common.PubkeyToHex(&s.alice.identity.PublicKey), community.ID(), request.AddressesToReveal)
+	s.Require().NoError(err)
+
 	passwdHash := types.EncodeHex(crypto.Keccak256([]byte(alicePassword)))
-	request := &requests.EditSharedAddresses{CommunityID: community.ID(), Password: passwdHash, AddressesToReveal: []string{aliceAddress1}, AirdropAddress: aliceAddress1}
+	for i := range signingParams {
+		signingParams[i].Password = passwdHash
+	}
+	signatures, err := s.alice.SignData(signingParams)
+	s.Require().NoError(err)
+
+	updateAddresses := len(request.AddressesToReveal) == 0
+	if updateAddresses {
+		request.AddressesToReveal = make([]string, len(signingParams))
+	}
+	for i := range signingParams {
+		request.AddressesToReveal[i] = signingParams[i].Address
+		request.Signatures = append(request.Signatures, types.FromHex(signatures[i]))
+	}
+	if updateAddresses {
+		request.AirdropAddress = request.AddressesToReveal[0]
+	}
```

Comment:
Maybe worth creating a helper func? I see this duplicated few times.

Reply:
Where are those few places? I know only for 2, but they are similar, not the same, cause one deals with `EditSharedAddresses` and another `RequestToJoinCommunity`.